src/MigrationGenerator.php

```diff
@@ -387,73 +384,57 @@ protected function generateMigrationPlugin(SourcePluginInterface $migration) {
    * @param string $target_entity_type
    *   The target entity type to reference.
-   * @param bool $multiple
-   *   Whether the reference is multiple or not.
    * @param string $field_type
    *   The field type to be referenced.
    */
-  protected function addTargetMigration(array &$migration_plugin, $dest_field, array $target_ids, $target_entity_type, $multiple, $field_type = 'entity_reference') {
+  protected function addTargetMigration(array &$migration_plugin, $dest_field, array $target_ids, $target_entity_type, $field_type = 'entity_reference') {
     $migrations = $this->getMigrations();
     $target_ids = array_intersect(array_keys($migrations), $target_ids);
     if (!empty($target_ids)) {
       $dest_subfield = explode('/', $dest_field);
       if (isset($dest_subfield[1]) && $dest_subfield[1] != 'target_id' && $dest_subfield[1] != 'target_revision_id') {
         return $migration_plugin;
       }
-      if ($multiple) {
-        $target_key = $this->targetMigrationKey($target_ids);
-
-        $process = [
-          'plugin' => 'iterator',
-          'process' => [
-            'target_id' => [
-              'plugin' => 'migration_lookup',
-              'migration' => $target_ids,
-              'source' => $target_key,
-              'no_stub' => TRUE,
-            ],
+      $target_key = $this->targetMigrationKey($target_ids);
+
+      $migration_plugin['process'][$dest_field][] = [
+        'plugin' => 'normalize_entity_reference',
+        'replacement_key' => $target_key,
+        'delimiter' => ',',
+      ];
+
+      $process = [
+        'plugin' => 'sub_process',
+        'process' => [
+          'target_id' => [
+            'plugin' => 'migration_lookup',
+            'migration' => $target_ids,
+            'source' => $target_key,
           ],
-        ];
+        ],
+      ];
 
-        // Add the rest of subproperties for this field.
-        $definition = \Drupal::service('plugin.manager.field.field_type')->getDefinition($field_type);
-        $field_definition = BaseFieldDefinition::create($field_type);
-        $schema = call_user_func([$definition['class'], 'schema'], $field_definition);
-        foreach (array_keys($schema['columns']) as $subproperty) {
-          if ($subproperty != 'target_id') {
-            $process['process'][$subproperty] = $subproperty;
-          }
-          // Add only for entity_reference revisions.
-          if ($subproperty == 'target_revision_id') {
-            $process['process']['target_revision_id'] = [
-              'plugin' => 'content_id_to_revision_reference',
-              'target_entity_type' => $target_entity_type,
-              'source' => '@target_id',
-              'lookup_id' => TRUE,
-            ];
-          }
+      // Add the rest of subproperties for this field.
+      $definition = \Drupal::service('plugin.manager.field.field_type')->getDefinition($field_type);
+      $field_definition = BaseFieldDefinition::create($field_type);
```

Comment:
I don't know how we can get the BaseFieldDefinition for a given $field_type with a service instead of a static method

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Field%21BaseFieldDefinition.php/function/BaseFieldDefinition%3A%3Acreate/8.6.x

Any ideas?

Reply:
Yeah..., I overlooked that you were creating a Field definition instead of a more generic entity type, where entityTypeManager is enough.
Let's use the static method... We have almost all the different pieces to make it through injection, but I believe it's not worth the effort and code looks cleaner in this way.